omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/EncounterResource1_8.java

```diff
@@ -206,6 +207,22 @@ public void delete(Encounter enc, String reason, RequestContext context) throws
 		Context.getEncounterService().voidEncounter(enc, reason);
 	}
 	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#unDelete(java.lang.Object,
+	 *      org.openmrs.module.webservices.rest.web.RequestContext)
+	 */
+	@Override
+	protected Object unDelete(Encounter enc, RequestContext context) {
+		Encounter unvoidedEnc = null;
+		if (!enc.isVoided()) {
+			unvoidedEnc = Context.getEncounterService().getEncounter(enc.getEncounterId());
```

Comment:
You already have "Encounter enc" passed into the method. Why do you need to fetch it again? Can't you just do
```
if (enc.isVoided()) {
    // unvoid it
}
return // convert
```

Reply:
Thank you for this observation, I have adjusted the code accordingly.